flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/history/HistoryServer.java

```diff
@@ -244,13 +246,41 @@ public HistoryServer(
                     "Cannot set %s to 0 or less than -1",
                     HistoryServerOptions.HISTORY_SERVER_RETAINED_JOBS.key());
         }
+        boolean remoteFetchEnabled =
+                config.contains(HistoryServerOptions.HISTORY_SERVER_CACHED_JOBS);
+
+        int generalCachedJobSize = -1;
+        if (remoteFetchEnabled) {
+            generalCachedJobSize = config.get(HistoryServerOptions.HISTORY_SERVER_CACHED_JOBS);
+            if (generalCachedJobSize == 0 || generalCachedJobSize < -1) {
+                throw new IllegalConfigurationException(
+                        "Cannot set %s to 0 or less than -1",
+                        HistoryServerOptions.HISTORY_SERVER_CACHED_JOBS.key());
+            }
+        }
+        int numCachedMostRecentlyViewedJobs =
+                config.get(
+                        HistoryServerOptions.HISTORY_SERVER_NUM_CACHED_MOST_RECENTLY_VIEWED_JOBS);
+        if (numCachedMostRecentlyViewedJobs <= 0) {
+            throw new IllegalConfigurationException(
+                    "Cannot set %s to less than 0",
+                    HistoryServerOptions.HISTORY_SERVER_NUM_CACHED_MOST_RECENTLY_VIEWED_JOBS.key());
+        }
```

Comment:
How about abstracting the logic for determining the values of these two parameters into one or two private methods?
That way, we may reduce some redundant code or improve code readability in different ways.